src/seedu/addressbook/data/person/Address.java

```diff
@@ -1,18 +1,32 @@
 package seedu.addressbook.data.person;
 
 import seedu.addressbook.data.exception.IllegalValueException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    // format: BLOCK, STREET, UNIT, POSTAL_CODE
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in the format of: " +
+            "'a/BLOCK, STREET, UNIT, POSTAL_CODE'";
+    public static final String ADDRESS_VALIDATION_REGEX = "(.+), (.+), (.+), (.+)";
 
+    private static final int BLOCK_INDEX = 1;
```

Comment:
2 things here:

- Good to see you have eliminated magic numbers in the code
- However, please take a look at the coding standards and find out what is the correct convention for naming constants